src/DateTimeParser.jl

```diff
@@ -0,0 +1,512 @@
+module DateTimeParser
+
+using Base.Dates
+using TimeZones
+import Base.Dates: VALUETODAYOFWEEK, VALUETODAYOFWEEKABBR, VALUETOMONTH, VALUETOMONTHABBR
+export parsedate
+
+# Automatic parsing of DateTime strings. Based upon Python's dateutil parser
+# https://labix.org/python-dateutil#head-a23e8ae0a661d77b89dfb3476f85b26f0b30349c
+
+# Some pointers:
+# http://www.cl.cam.ac.uk/~mgk25/iso-time.html
+# http://www.w3.org/TR/NOTE-datetime
+# http://new-pds-rings-2.seti.org/tools/time_formats.html
+# http://search.cpan.org/~muir/Time-modules-2003.0211/lib/Time/ParseDate.pm
+
+const HMS = Dict{AbstractString, Int}(
+    "h" => 1, "hour" => 1, "hours" => 1,
+    "m" => 2, "minute" => 2, "minutes" => 2,
+    "s" => 3, "second" => 3, "seconds" => 3,
+)
+const AMPM = Dict{AbstractString, Int}(
+    "am" => 1, "a" => 1,
+    "pm" => 2, "p" => 2,
+)
+const JUMP = (
+    " ", ".", ",", ";", "-", "/", "'", "at", "on", "and", "ad", "m", "t", "of", "st",
+    "nd", "rd", "th", "the",
+)
+const PERTAIN = ("of",)
+const UTCZONE = ("utc", "gmt", "z",)
+
+function parsedate(datetimestring::AbstractString, fuzzy::Bool=false;
+    default::ZonedDateTime=ZonedDateTime(DateTime(today()),localzone()),
+    timezone_infos::Dict{AbstractString, TimeZone} = Dict{AbstractString, TimeZone}(), # Specify what a timezone is
+    dayfirst::Bool=false, # MM-DD-YY vs DD-MM-YY
+    yearfirst::Bool=false, # MM-DD-YY vs YY-MM-DD
+    language::AbstractString="english", # Language in Dates.VALUETOMONTH and VALUETODAYOFWEEK
+)
+    datetimestring = strip(datetimestring)
+
+    if isempty(datetimestring)
+        return default
+    end
+
+    weekday = Dict{AbstractString, Int}()
```

Comment:
Types here should match what is in `Base.Dates`. Use `UTF8String` for these Dict definitions rather than `AbstractString`